src/main/java/uk/ac/ic/wlgitbridge/snapshot/base/Request.java

```diff
@@ -73,9 +74,30 @@ private T getResult() throws MissingRepositoryException, FailedConnectionExcepti
             if (cause instanceof HttpResponseException) {
                 HttpResponseException httpCause = (HttpResponseException) cause;
                 int sc = httpCause.getStatusCode();
-                if (sc == HttpServletResponse.SC_UNAUTHORIZED || sc == HttpServletResponse.SC_FORBIDDEN) {
+                if (sc == HttpServletResponse.SC_UNAUTHORIZED || sc == HttpServletResponse.SC_FORBIDDEN) {  // 401, 403
                     throw new ForbiddenException();
-                } else if (sc == HttpServletResponse.SC_NOT_FOUND) {
+                } else if (sc == HttpServletResponse.SC_CONFLICT) {  // 409
+                    try {
+                        JsonObject json = Instance.gson.fromJson(httpCause.getContent(), JsonObject.class);
+                        String code = json.get("code").getAsString();
+                        if ("projectHasDotGit".equals(code)) {
+                            throw new MissingRepositoryException(Arrays.asList(
+                                    "This project contains a '.git' entity at the top level, indicating that it is",
+                                    "already a git repository. The Overleaf git-bridge cannot work with this project",
+                                    "due to a known problem with handling these '.git' folders.",
+                                    "",
+                                    "If this is unexpected, please contact us at support@overleaf.com, or",
```

Comment:
Not essential, but maybe in this case it would be better to tell people to remove the `.git` folder from their project and try again? I'm not sure we need to direct them to support.

Reply:
Good call, I'll patch that

Reply:
New PR: https://github.com/overleaf/writelatex-git-bridge/pull/66